Automattic/woocommerce-payments
# Fix pagination issue with dispute count in task list on overview page

Fixes https://github.com/Automattic/woocommerce-payments/issues/4123.

#### Changes proposed in this Pull Request

On Payments > Overview page, we display a task to open the awaiting response dispute list under the `Things to do` list. Before this PR, we're getting a list of disputes and count the ones that need response to get the number in the task. This PR takes a different approach by calling a REST API to get a list of dispute statuses and their counts, solving the problem where awating response disputes got buried in the page 2+.

#### Testing instructions





0. Make sure to enable `Enable account overview task list` in WCPay Dev because the task list / 'Things to do' is behind a feature flag.
1. **You will need >25 disputed WCPay transactions**. Do this by purchasing products multiple times with either of the following cards (tip: save the cards when checking out to reduce repetitive card number form inputs) :
	* `4000000000001976`
	* `4000000000000259`
2. Checkout the `trunk` branch.
3. Visit **Payments → Overview** and observe the disputes task in the **Things to do** task list. Notice that the # of disputed payments is **incorrect**. An easy way to check is to compare with the Disputes badge in the admin menu:

   <img src="https://user-images.githubusercontent.com/3147296/177670410-4b49f69c-dd67-4b3d-9f10-7341626c20bf.png" alt="before" width="500" />
5. Checkout this PR branch `fix/4123-dispute-count`  and run `npm run build:client`
6. Visit **Payments → Overview** and observe the disputes task in the **Things to do** task list. Notice that the # of disputed payments **correct**:

   <img src="https://user-images.githubusercontent.com/3147296/177670390-8b21d688-4abf-4fc4-b14a-ff906f0fa37b.png" alt="after" width="500" />

#### Notes
- Use nodejs 14 as I find problems running `npm run test:js` locally with nodejs 16.

-------------------


**Post merge**
